src/main/java/org/hiero/sketch/table/PartialMembershipDense.java

```diff
@@ -65,6 +73,81 @@ public int getSize() {
     }
 
     @Override
+    public int getSize(boolean exact) {
+        if (this.rowCountCorrect)
+            return this.rowCount;
+        if (exact)
+            return this.getSize();
+        IMembershipSet sampleSet = this.sample(20);
+        int snumber = 0;
+        IRowIterator it = sampleSet.getIterator();
+        int curr = it.getNextRow();
+        while (curr >= 0) {
+            if (this.filter.test(curr))
+                snumber++;
+            curr = it.getNextRow();
+        }
+        return (this.baseMap.getSize(false) * snumber) / sampleSet.getSize(true);
+    }
+
+    /**
+     * @return A sample of k items from the membership set. The sample is with replacement so may
+     * contain less than k distinct elements. The sample is obtained by sampling k items from the
+     * base map and filtering it. This is done 10 times, if k samples had been found the function
+     * gives up and returns whatever was found. This will happen if the membership is sparse.
+     */
+    @Override
+    public IMembershipSet sample(int k) {
+        int samples = 0;
+        IMembershipSet batchSet;
+        IntOpenHashSet sampleSet = new IntOpenHashSet();
+        for (int attempt = 0; attempt < 10; attempt++) {
+            batchSet = this.baseMap.sample(k);
+            IRowIterator it = batchSet.getIterator();
+            int tmprow = it.getNextRow();
+            while (tmprow >= 0) {
+                if (isMember(tmprow)) {
+                    sampleSet.add(tmprow);
+                    samples++;
+                    if (samples == k)
+                        return new PartialMembershipSparse(sampleSet);
+                }
+                tmprow = it.getNextRow();
+            }
+        }
+        return new PartialMembershipSparse(sampleSet);
+    }
+
+    /**
+     * Samples the base map for k items and then applies the filter on that set. Makes 10 attempts to reach k samples
+     * this way and then gives up and returns whatever was sampled.
+     * @param k
```

Comment:
If you don't want to document the parameters please remove them from the comment.